src/include/stir/recon_buildblock/KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData.h

```diff
@@ -0,0 +1,266 @@
+
+  //! Destructor
+  /*! Calls end_distributable_computation()
+   */
+  ~KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData();
+
+  /*! \name Functions to get parameters
+   \warning Be careful with changing shared pointers. If you modify the objects in
+   one place, all objects that use the shared pointer will be affected.
+  */
+  //@{
+
+  //kernel
+  const std::string get_anatomical_filename() const;
+  const int get_num_neighbours() const;
+  const int get_num_non_zero_feat() const;
+  const double get_sigma_m() const;
+  double get_sigma_p();
+  double get_sigma_dp();
+  double get_sigma_dm();
+  const bool get_only_2D() const;
+  int get_subiter_num();
+  double get_kSD();
+  bool get_hybrid();
+
+   shared_ptr<TargetT>& get_kpnorm_sptr();
+   shared_ptr<TargetT>& get_kmnorm_sptr();
+   shared_ptr<TargetT>& get_anatomical_sptr();
+
+  //@}
+  /*! \name Functions to set parameters
+    This can be used as alternative to the parsing mechanism.
+   \warning After using any of these, you have to call set_up().
+   \warning Be careful with changing shared pointers. If you modify the objects in
+   one place, all objects that use the shared pointer will be affected.
+
+  */
+  //@{
+
+  void set_subiter_num(int subiter_num);
+  void set_kSD(double kSD);
+  void set_kpnorm_sptr(shared_ptr<TargetT>&);
+  void set_kmnorm_sptr(shared_ptr<TargetT>&);
+  void set_anatomical_sptr(shared_ptr<TargetT>&);
+
+  //@}
+
+  virtual void
+    compute_sub_gradient_without_penalty_plus_sensitivity(TargetT& gradient,
+                                                          const TargetT &current_estimate,
+                                                          const int subset_num);
+protected:
+
+  virtual double
+    actual_compute_objective_function_without_penalty(const TargetT& current_estimate,
+                                                      const int subset_num);
+
+protected:
+  //! Filename with input projection data
+  std::string input_filename,kernelised_output_filename_prefix;
+  std::string current_kimage_filename;
+  std::string sens_filenames;
+ int subiter_num;
+
+  //! Anatomical image filename
+ std::string anatomical_image_filename;
+  mutable Array<3,float> distance;
+  double kSt_dev;
+  shared_ptr<TargetT> anatomical_sptr;
+  shared_ptr<TargetT> kpnorm_sptr,kmnorm_sptr;
+ //kernel parameters
+  int num_neighbours,num_non_zero_feat,num_elem_neighbourhood,num_voxels,dimz,dimy,dimx;
+  double sigma_m;
+  bool only_2D;
+  bool hybrid;
+  double sigma_p;
+  double sigma_dp, sigma_dm;
+
+
+  //! sets any default values
+  /*! Has to be called by set_defaults in the leaf-class */
+  virtual void set_defaults();
+  //! sets keys for parsing
+  /*! Has to be called by initialise_keymap in the leaf-class */
+  virtual void initialise_keymap();
+  //! checks values after parsing
+  /*! Has to be called by post_processing in the leaf-class */
+  virtual bool post_processing();
+
```

Comment:
add a `set_up()`

Reply:
@KrisThielemans  I added set_up as follow:
```
template<typename TargetT>
Succeeded
KernelisedPoissonLogLikelihoodWithLinearModelForMeanAndProjData<TargetT>::
set_up(shared_ptr<TargetT>  const& target_sptr)
{
    base_type::set_up(target_sptr);

    PoissonLogLikelihoodWithLinearModelForMeanAndProjData<TargetT>::set_up (target_sptr);
    this->subiteration_counter=0;
    this->anatomical_sd=0;
    return Succeeded::yes;
}
```
but when I run the images are all zeros